xblock/mixins.py

```diff
@@ -610,3 +611,43 @@ def has_support(self, view, functionality):
             True or False
         """
         return hasattr(view, "_supports") and functionality in view._supports  # pylint: disable=protected-access
+
+
+class CompletableXBlockMixin(object):
+    """
+    This mixin sets attributes and provides helper method to integrate XBlock with Completion API.
+    """
+
+    has_custom_completion = True
+    completion_method = 'completable'
+
+    def emit_completion(self, completion_ratio):
+        """
+        Emits completion event through Completion API.
+
+        Unlike grading API, calling this method allows completion to go down - i.e. emitting a value of 0.0 on
+        a previously completed block indicates that it is no longer considered complete.
+
+        Arguments:
+            completion_ratio (float): Ratio of completion within [0.0; 1.0] range (inclusive). 0.0 means the block
+            is not completed, 1.0 means the block is fully completed.
+
+        Returns:
+            None
+        """
+        if not self.has_custom_completion or self.completion_method != 'completable':
+            raise AttributeError(
+                "Using `emit_completion` requires `has_custom_completion == True` (was {}) "
+                "and `completion_method == 'completable'` (was {})".format(
+                    self.has_custom_completion, self.completion_method
+                )
+            )
+
+        if completion_ratio is None or not 0.0 <= completion_ratio <= 1.0:
```

Comment:
Can you drop `completion_ratio is None or` ?

Reply:
@robrap It'll cause python 3 tests to fail with TypeError or something like that. Basically python 3 is more picky when it comes to comparing None to anything else.

Reply:
Got it.  Thank you.